### PR TITLE
Use networking instead of vsock in launcher, stage1 and sysimage

### PR DESCRIPTION
**Pull request by andrisaar**
This has a "v0.5" feel to it; it's somewhat intentional as it's supposed to be (mostly) proof-of-concept.

What this PR does is modify the launcher and stage1 to use networking instead of vsock, and then enables QEMU user-mode networking so that stage1 can load the system image from the launcher.

This is still in the v0.5 territory as in the future we will want to use a proper TAP interface; however, doing it this way means you don't need `root` to run QEMU. The addresses used (10.0.2.2 is always the host, 10.0.2.10 is the guest) are what QEMU uses by default.

There is still one snag in there: `systemd-networkd` thinks that eth0 is in `pending` state, and thus it takes ~2 minutes before it times out and the orchestrator is started. I need to figure out what's going on with `systemd-networkd` in there, as the network interface itself is up and running, and thus the orchestrator _should_ be able to talk to 10.0.2.2:8080 just fine.

Also: apologies to @jul-sh but I didn't change the orchestrator to use networking; however, the change should be ~equivalent to the one I did in stage1.

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]
Nice.

How long does it take to load the system image over this network?

**Comment by andrisaar** [comment]
> Nice.
> 
> How long does it take to load the system image over this network?

I don't have concrete measurements; it does _feel_ a bit longer, but I reckon it's somewhere around 30-40% longer, not multiple times longer.

File: oak_containers_launcher/src/lib.rs [hunk 1]
```diff
@@ -47,16 +46,16 @@ pub fn path_exists(s: &str) -> Result<std::path::PathBuf, String> {
...
     let server = oak_containers_launcher_server::new(
-        args.vsock_cid,
-        args.vsock_port,
+        sockaddr,
         args.system_image,
         args.container_bundle,
         args.application_config,
     );
 
     // Use our PID for the CID of the guest.
```
Comment (jul-sh) [inline oak_containers_launcher/src/lib.rs:57]: comment no longer applies :)

**Review by jul-sh — APPROVED** [review]
nice :)

**Pull request merged**